### PR TITLE
threadpool: send signal only when queue is empty v2

### PR DESCRIPTION
**Pull request by chenttuuvv**
@saghul please review this patch

### CODE REVIEW

File: src/threadpool.c [hunk 1]
```diff
@@ -101,9 +101,12 @@ static void worker(void* arg) {
 
 
 static void post(QUEUE* q) {
+  int empty_queue;
   uv_mutex_lock(&mutex);
+  empty_queue = QUEUE_EMPTY(&wq) ? 1 : 0;
```
Comment (saghul) [inline src/threadpool.c]: You don't need to use the ternary operator, assigning QUEUE_EMPTY(x) will do.

**Comment by saghul** [comment]
Small nit, LGTM otherwise if the CI is happy (it seemed so).

I'll be away for a few days, I'll check the CI results more thoroughly and land it if all is good on Monday.

**Comment by saghul** [comment]
@bnoordhuis since you wrote the threadpool, are you onboard here? (just double checking I'm not missing anything)

**Comment by bnoordhuis** [comment]
LGTM sans the nit @saghul pointed out.

**Comment by saghul** [comment]
For reference, #460 landed.

**Pull request closed without merging**